ext/bg/js/translator.js

```diff
@@ -158,21 +266,33 @@ class Translator {
     }
 
     async buildTermFrequencies(definition, titles) {
-        definition.frequencies = [];
-        for (const meta of await this.database.findTermMeta(definition.expression, titles)) {
-            if (meta.mode === 'freq') {
-                definition.frequencies.push({
-                    expression: meta.expression,
-                    frequency: meta.data,
-                    dictionary: meta.dictionary
-                });
+        let terms = [];
+        if (definition.expressions) {
+            terms = terms.concat(definition.expressions);
+        } else {
+            terms.push(definition);
+        }
+
+        for (const term of terms) {
+            term.frequencies = [];
+            for (const meta of await this.database.findTermMeta(term.expression, titles)) {
+                if (meta.mode === 'freq') {
+                    term.frequencies.push({
+                        expression: meta.expression,
+                        frequency: meta.data,
+                        dictionary: meta.dictionary
+                    });
+                }
             }
         }
     }
 
     async expandTags(names, title) {
         const tags = [];
         for (const name of names) {
+            if (typeof name !== 'string') {
```

Comment:
What else could `name` be?

Reply:
This is related to the issue with `dictTermsMergeByGloss`. Some of the definitions come from `Translator.findTerms` and have their tags expanded while others don't. A better way to handle this would probably be to check the type before even calling `expandTags`.

Reply:
Same comment as the one about the tags above. You should always know what tags are, and it should be possible to infer what type the variable is just from conventions in the code.